src/lib.rs

```diff
@@ -106,51 +110,120 @@ fn validate(payload: &[u8]) -> CallResult {
     }
 }
 
-fn get_all_container_images(pod: apicore::Pod) -> Vec<String> {
-    let mut vec: Vec<String> = Vec::new();
-    if let Some(spec) = pod.spec {
-        vec.append(
-            &mut spec
-                .containers
-                .into_iter()
-                .filter_map(|x| x.image)
-                .collect(),
-        );
-        if let Some(init_containers) = spec.init_containers {
-            vec.append(
-                &mut init_containers
-                    .into_iter()
-                    .filter_map(|x| x.image)
-                    .collect(),
-            );
-        }
-        if let Some(ephemeral_containers) = spec.ephemeral_containers {
-            vec.append(
-                &mut ephemeral_containers
-                    .into_iter()
-                    .filter_map(|x| x.image)
-                    .collect(),
-            );
+// verify all images and return a PodSpec with the images replaced with the digest which was used for the verification
+fn verify_all_images_in_pod(
+    spec: &PodSpec,
+    policy_verification_errors: &mut Vec<String>,
+    signatures: &[Signature],
+) -> PodSpec {
```

Comment:
Why not return a `Result<PodSpec>` instead? and then use a match in `validate()`

Reply:
great idea! done in https://github.com/kubewarden/verify-image-signatures/pull/4/commits/3f8ff915481f74266e13c89fa7c0beb0e65a5ed8